components/Collections/src/MDCCollectionViewController.m

```diff
@@ -298,6 +301,29 @@ - (CGSize)inlaidSizeAtIndexPath:(NSIndexPath *)indexPath withSize:(CGSize)size {
   return size;
 }
 
+#pragma mark - Subclassing Methods
+
+/*
+ The below method is solely used for subclasses to retrieve width information in order to
+ calculate cell height. Not meant to call method cellWidthAtSectionIndex:collectionView as
+ that method recalculates section insets which we don't want to do.
+ */
+- (CGFloat)cellWidthAtSectionIndex:(NSInteger)section {
+  CGFloat bounds = CGRectGetWidth(
+                                  UIEdgeInsetsInsetRect(self.collectionView.bounds,
+                                                        self.collectionView.contentInset));
+  UIEdgeInsets sectionInsets = [self collectionView:self.collectionView
+                                             layout:self.collectionView.collectionViewLayout
+                             insetForSectionAtIndex:section];
+
+  CGFloat insets = sectionInsets.left + sectionInsets.right;
+  if (_styler.cellLayoutType == MDCCollectionViewCellLayoutTypeGrid) {
```

Comment:
MDCCollectionViewCellLayoutTypeList is the equivalent of `nil`, so it will skip this conditional and return the default.  

I would suggest that anyone setting a Grid style with no columns in the grid wouldn't be surprised that the cells have an invalid width. I would support a check (and maybe an NSAssert to warn devs) if the styler is in an inconsistent state, but I think the div-by-zero should be pretty obvious as well.